include/avif/avif.h

```diff
@@ -28,10 +36,19 @@ extern "C" {
 // For static build, AVIF_API is always defined as nothing.
 
 #if defined(_WIN32)
+#if defined(AVIF_USE_NODISCARD) && (defined(__GNUC__) || defined(__clang__))
+#define AVIF_HELPER_EXPORT [[gnu::dllexport]]
+#define AVIF_HELPER_IMPORT [[gnu::dllimport]]
```

Comment:
These two GCC attributes are document at https://gcc.gnu.org/onlinedocs/gcc/Microsoft-Windows-Function-Attributes.html.

Reply:
Aren't these considered valid only with C23 or gnu17?

https://gcc.gnu.org/onlinedocs/gcc/Attribute-Syntax.html#Attribute-Syntax

Reply:
The definition of the `AVIF_USE_NODISCARD` macro at line 19 detects C++17 or C23:

```
#if defined(AVIF_ENABLE_NODISCARD) || (defined(__cplusplus) && __cplusplus >= 201703L) || \
    (defined(__STDC_VERSION__) && __STDC_VERSION__ >= 202311L)
#define AVIF_USE_NODISCARD 1
#endif
```

C++17 is the first C++ version that supports the `nodiscard` attribute.

C23 is the first C version that supports the standard `[[...]]` attribute syntax.

Reply:
I'm sorry I missed the use of `AVIF_USE_NODISCARD` in this section and only read the second half.